tests/unit/plugins/modules/files/test_archive.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.general.tests.unit.compat.mock import Mock, patch
+from ansible_collections.community.general.tests.unit.plugins.modules.utils import ModuleTestCase, set_module_args
+from ansible_collections.community.general.plugins.modules.files.archive import get_archive
+
+
+class TestArchive(ModuleTestCase):
+    def setUp(self):
+        super(TestArchive, self).setUp()
+
+        self.mock_os_path_isdir = patch('os.path.isdir')
+        self.os_path_isdir = self.mock_os_path_isdir.start()
+
+    def tearDown(self):
+        self.os_path_isdir = self.mock_os_path_isdir.stop()
+
+    def test_archive_root(self):
+        set_module_args(
+            dict(
+                path=['/foo', '/bar', '/baz', '/foobar', '/barbaz', '/foo/bar'],
+                dest='/destination.tgz'
+            )
+        )
+
+        module = AnsibleModule(
```

Comment:
How about moving that code in the module to a function so it doesn't have to be replicated here?